config/config.go

```diff
@@ -175,47 +179,92 @@ func createConfigFromMetadata(md metadataJSON) *config {
 	}
 
 	// Check project first then instance as instance metadata overrides project.
-	if md.Project.Attributes.InventoryEnabledOld != "" {
-		c.osInventoryEnabled = parseBool(md.Project.Attributes.InventoryEnabledOld)
-	}
-	if md.Project.Attributes.InventoryEnabled != "" {
+	switch {
+	case md.Project.Attributes.InventoryEnabled != "":
 		c.osInventoryEnabled = parseBool(md.Project.Attributes.InventoryEnabled)
+	case md.Project.Attributes.InventoryEnabledOld != "":
+		c.osInventoryEnabled = parseBool(md.Project.Attributes.InventoryEnabledOld)
 	}
+
+	c.parseFeatures(md.Project.Attributes.PreReleaseFeaturesOld, true)
 	c.parseFeatures(md.Project.Attributes.PreReleaseFeatures, true)
-	c.parseFeatures(md.Project.Attributes.EnabledFeatures, true)
+	if md.Project.Attributes.OSConfigEnabled != "" {
+		e := parseBool(md.Project.Attributes.OSConfigEnabled)
+		c.taskNotificationEnabled = e
+		c.guestPoliciesEnabled = e
+		c.osInventoryEnabled = e
+	}
 	c.parseFeatures(md.Project.Attributes.DisabledFeatures, false)
 
-	if md.Instance.Attributes.InventoryEnabledOld != "" {
-		c.osInventoryEnabled = parseBool(md.Instance.Attributes.InventoryEnabledOld)
-	}
-	if md.Instance.Attributes.InventoryEnabled != "" {
+	switch {
+	case md.Instance.Attributes.InventoryEnabled != "":
 		c.osInventoryEnabled = parseBool(md.Instance.Attributes.InventoryEnabled)
+	case md.Instance.Attributes.InventoryEnabledOld != "":
+		c.osInventoryEnabled = parseBool(md.Instance.Attributes.InventoryEnabledOld)
 	}
+
+	c.parseFeatures(md.Instance.Attributes.PreReleaseFeaturesOld, true)
 	c.parseFeatures(md.Instance.Attributes.PreReleaseFeatures, true)
-	c.parseFeatures(md.Instance.Attributes.EnabledFeatures, true)
+	if md.Instance.Attributes.OSConfigEnabled != "" {
+		e := parseBool(md.Instance.Attributes.OSConfigEnabled)
+		c.taskNotificationEnabled = e
+		c.guestPoliciesEnabled = e
+		c.osInventoryEnabled = e
+	}
 	c.parseFeatures(md.Instance.Attributes.DisabledFeatures, false)
 
+	if md.Instance.Attributes.PollIntervalOld != nil {
+		if val, err := md.Instance.Attributes.PollIntervalOld.Int64(); err == nil {
+			c.osConfigPollInterval = int(val)
+		}
+	}
+	if md.Instance.Attributes.PollIntervalOld != nil {
+		if val, err := md.Instance.Attributes.PollIntervalOld.Int64(); err == nil {
+			c.osConfigPollInterval = int(val)
+		}
+	}
+
 	if md.Instance.Attributes.PollInterval != nil {
 		if val, err := md.Instance.Attributes.PollInterval.Int64(); err == nil {
 			c.osConfigPollInterval = int(val)
 		}
 	}
+	if md.Instance.Attributes.PollInterval != nil {
+		if val, err := md.Instance.Attributes.PollInterval.Int64(); err == nil {
+			c.osConfigPollInterval = int(val)
+		}
+	}
+
+	switch {
+	case md.Instance.Attributes.DebugEnabledOld != "":
```

Comment:
Not project first then instance here and for the endpoint setting?

Reply:
Yep, missed, I updated and cleaned up the code more